app/workers/revert_application_choices_updated_at_worker.rb

```diff
@@ -0,0 +1,48 @@
+class RevertApplicationChoicesUpdatedAtWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :low_priority
+
+  def perform(choice_ids)
+    choice_ids_to_update = {}
+
+    ApplicationChoice.where(id: choice_ids).find_each(batch_size: 100) do |choice|
+      audits = choice.own_and_associated_audits
+      audits_we_made = audits
+        .where(created_at: DateTime.new(2024, 9, 3, 11)..DateTime.new(2024, 9, 3, 20))
+        .where(user_type: nil, user_id: nil)
+        .where(action: :create)
+        .where(username: '(Automated process)')
+        .where(associated_type: 'ApplicationChoice')
+        .where(auditable_type: %w[ApplicationExperience ApplicationWorkHistoryBreak])
+        .order(id: :asc)
+
+      if audits_we_made.blank?
+        Rails.logger.tagged('Big Touch') do
+          Rails.logger.info("Has no touch audits choice id #{choice.id}")
+        end
+        next
+      end
+
+      audits_without_the_ones_we_made = audits.where.not(id: audits_we_made.select(:id))
+      last_created_at_from_audits = audits_without_the_ones_we_made.maximum(:created_at)
+
+      if last_created_at_from_audits.before?(audits_we_made.minimum(:created_at))
+        choice_ids_to_update[choice.id] = last_created_at_from_audits
+      else
+        Rails.logger.tagged('Big Touch') do
+          Rails.logger.info("User audit created at > audits_we_made  #{choice.id}")
+        end
+      end
+    end
+
+    ActiveRecord::Base.transaction do
+      ApplicationChoice.where(id: choice_ids_to_update.keys).each do |choice|
+        Rails.logger.tagged('Big Touch') do
+          Rails.logger.info("Update choice id #{choice.id}")
+        end
+        choice.update_columns(updated_at: choice_ids_to_update[choice.id])
+      end
+    end
```

Comment:
```suggestion
    updates_for_upserting = choice_ids_to_update.map { |id, updated_at| { id: id, updated_at: updated_at } }

ApplicationChoice.upsert_all(
  updates_for_upserting,
  unique_by: :id,
  update_only: [:updated_at],
  record_timestamps: false
)
```

I never know how `upsert_all` works and not had much success with it in the past but this "in theory' would update all ApplicationChoices with the appropriate `updated_at` all in one SQL execution 🤞
Would be worth trialling this to see how much quicker it makes this whole process.

Reply:
I did think about upsert_all but I don't think it's worth it. The performance gain is there but I'm not sure if it's worth the risk of inserting records by mistake.

I think we would prefer to just update them one by one but in a transaction.